src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -3250,6 +3250,99 @@ class AlterStageSegment(BaseSegment):
     )
 
 
+@snowflake_dialect.segment()
+class AppendOnlyExpressionSegment(BaseSegment):
+    """An APPEND_ONLY = TRUE | FALSE expression."""
+
+    type = "append_only_expression"
+    match_grammar = "APPEND_ONLY"
+
+    parse_grammar = Sequence(
+        "APPEND_ONLY",
+        Ref("EqualsSegment"),
+        OneOf(Ref("TrueSegment"), Ref("FalseSegment"))
+    )
+
+
+@snowflake_dialect.segment()
+class ShowInitialRowsExpressionSegment(BaseSegment):
+    """A SHOW_INITIAL_ROWS = TRUE | FALSE expression."""
+
+    type = "show_initial_rows_expression"
+    match_grammar = "SHOW_INITIAL_ROWS"
+
+    parse_grammar = Sequence(
+        "SHOW_INITIAL_ROWS",
+        Ref("EqualsSegment"),
+        OneOf(Ref("TrueSegment"), Ref("FalseSegment"))
+    )
+
+
+@snowflake_dialect.segment()
+class InsertOnlyExpressionSegment(BaseSegment):
+    """A INSERT_ONLY = TRUE expression."""
+
+    type = "insert_only_expression"
+    match_grammar = "INSERT_ONLY"
+
+    parse_grammar = Sequence(
+        "INSERT_ONLY",
+        Ref("EqualsSegment"),
+        Ref("TrueSegment")
+    )
+
+
+@snowflake_dialect.segment()
+class CreateStreamStatementSegment(BaseSegment):
+    """A Snowflake `CREATE STREAM` statement.
+
+    https://docs.snowflake.com/en/sql-reference/sql/create-stream.html
+    """
+
+    type = "create_stream_statement"
+
+    match_grammar = Sequence(
+        "CREATE", "STREAM"
+    )
+
+    parse_grammar = Sequence(
+        "CREATE",
+        Ref("OrReplaceGrammar", optional=True),
+        "STREAM",
+        Ref("IfNotExistsGrammar", optional=True),
```

Comment:
This is the problem.

This says match `CREATE STREAM` and then **just that match** with the parse grammar. Obviously the statement includes more than just that, so this isn't enough in `match_grammar`.

Typically we use `StartsWith` in `match_grammar` and then have a `terminator` argument to that.

However, having a separate `match_grammar` and `parse_grammar` is only useful if you can prune part of the SQL (using `match_grammar`) to make the parsing easier.

Here you're not doing that, so not need to have them separated and can just combine them into a joint `match_grammar`.

```suggestion
    match_grammar = Sequence(
        "CREATE",
        Ref("OrReplaceGrammar", optional=True),
        "STREAM",
        Ref("IfNotExistsGrammar", optional=True),
```